storage/localstore/mode_set_test.go

```diff
@@ -63,44 +63,167 @@ func TestModeSetAccess(t *testing.T) {
+			mode:            chunk.ModeSetSyncPull,
+			anonymous:       true,
+			pin:             false,
+			runGcIndexTest:  false,
+			expErrPushIndex: leveldb.ErrNotFound,
+			expErrGCIndex:   nil,
+			expErrPinIndex:  leveldb.ErrNotFound,
+		},
+		{
+			name:            "set pull sync, anonymous tag, with pinning",
+			mode:            chunk.ModeSetSyncPull,
+			anonymous:       true,
+			pin:             true,
+			runGcIndexTest:  false,
+			expErrPushIndex: leveldb.ErrNotFound,
+			expErrGCIndex:   nil,
+			expErrPinIndex:  nil,
+		},
+		{
+			name:            "set push sync, normal tag, no pinning",
+			anonymous:       false,
+			pin:             false,
+			mode:            chunk.ModeSetSyncPush,
+			runGcIndexTest:  true,
+			expErrPushIndex: leveldb.ErrNotFound,
+			expErrGCIndex:   nil,
+			expErrPinIndex:  leveldb.ErrNotFound,
+		},
+		{
+			name:            "set push sync, normal tag, with pinning",
+			anonymous:       false,
+			pin:             true,
+			mode:            chunk.ModeSetSyncPush,
+			runGcIndexTest:  true,
+			expErrPushIndex: leveldb.ErrNotFound,
+			expErrGCIndex:   nil,
+			expErrPinIndex:  nil,
+		},
+		{
+			name:            "set push sync, anonymous tag, no pinning",
+			anonymous:       true,
+			pin:             false,
+			mode:            chunk.ModeSetSyncPush,
+			runGcIndexTest:  true,
+			expErrPushIndex: leveldb.ErrNotFound,
+			expErrGCIndex:   nil,
+			expErrPinIndex:  leveldb.ErrNotFound,
+		},
+		{
+			name:            "set push sync, anonymous tag, with pinning",
+			anonymous:       true,
+			pin:             true,
+			mode:            chunk.ModeSetSyncPush,
+			runGcIndexTest:  true,
+			expErrPushIndex: leveldb.ErrNotFound,
+			expErrGCIndex:   nil,
+			expErrPinIndex:  nil,
+		},
+	} {
+		t.Run(mtc.name, func(t *testing.T) {
+			for _, tc := range multiChunkTestCases {
+				t.Run(tc.name, func(t *testing.T) {
+					db, cleanupFunc := newTestDB(t, &Options{Tags: chunk.NewTags()})
+					defer cleanupFunc()
+
+					tag, err := db.tags.Create(mtc.name, int64(tc.count), mtc.anonymous)
+					if err != nil {
+						t.Fatal(err)
+					}
+					if tag.Uid != 0 {
+						t.Fatal("expected mock tag uid")
+					}
 
-			chunks := generateTestRandomChunks(tc.count)
+					chunks := generateTestRandomChunks(tc.count)
 
-			wantTimestamp := time.Now().UTC().UnixNano()
-			defer setNow(func() (t int64) {
-				return wantTimestamp
-			})()
+					wantTimestamp := time.Now().UTC().UnixNano()
+					defer setNow(func() (t int64) {
+						return wantTimestamp
+					})()
 
-			_, err := db.Put(context.Background(), chunk.ModePutUpload, chunks...)
-			if err != nil {
-				t.Fatal(err)
-			}
+					_, err = db.Put(context.Background(), chunk.ModePutUpload, chunks...)
+					if err != nil {
+						t.Fatal(err)
+					}
 
-			err = db.Set(context.Background(), chunk.ModeSetSync, chunkAddresses(chunks)...)
-			if err != nil {
-				t.Fatal(err)
-			}
+					tag.Total = int64(tc.count)
```

Comment:
Add a comment explaining why the test case parameter is injected into the object that is validate by the same test. Here counter are set explicitly and tagCounterTest takes the same value and tag object for validation. Function name tagCounterTest would suggest that the same is checked, but that function is doing a more specific check on synced counter.